pallets/liquidity-pools/src/lib.rs

```diff
@@ -897,24 +944,60 @@ pub mod pallet {
 					investor,
 					currency,
 					amount,
-				} => Self::handle_decrease_redemption(
+				} => Self::handle_decrease_redeem_order(
 					pool_id,
 					tranche_id,
 					investor.into(),
-					currency.into(),
 					amount,
+					currency.into(),
 					sender,
 				),
 				Message::CollectInvest {
 					pool_id,
 					tranche_id,
 					investor,
-				} => Self::handle_collect_investment(pool_id, tranche_id, investor.into()),
+					currency,
+				} => Self::handle_collect_investment(
+					pool_id,
+					tranche_id,
+					investor.into(),
```

Comment:
@wischli Only am making my comment here as this is a new line, but this refers to all locations where `[u8;32]` is transformed into `AccountId`. 

We must use a trait that is able to transfrom `(Domain, [u8;32]) -> AccountId`. Something like `trait Convert<(Domain, [u8;32], AccountId)>`.

The reason is that we do not want other domains to control the investments from the same account of another domain.

Reply:
We can then use the `runtime_common::conversion` logic for that. So it should be a an associated type on the LP config

Reply:
Please let me know if ff07d88207dcbbf46def8cea4f2cd75d1112a585 suffices.